sync `epic/ia` with `trunk`

jaredrethman opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Sync `epic/ia` with `trunk`. Changes (introduced in #3582) to Wizards needed to be migrated to new locations, some notable changes (13d6d55f6a3949ed17118a98369525cfc12ab442):
* Endpoint & callback for route `DELETE:/wizard/newspack-reader-revenue-wizard/reader-activation/emails/(?P<id>\d+)`  moved from `includes/wizards/class-reader-revenue-wizard.php` to `includes/wizards/audience/class-audience-donations.php`.
* Endpoint & callback for route `DELETE:/wizard/newspack-engagement-wizard/reader-activation/emails/(?P<id>\d+)` moved from `includes/wizards/class-engagement-wizard.php` to `includes/wizards/audience/class-audience-wizard.php`.
* Order notes checkbox added to `<BillingFields />` in `src/wizards/audience/components/billing-fields/index.tsx`
* "Checkout Configuration" group of components moved to Audience > Setup > Checkout & Payment.
* Audience > Setup tabs got wrapped in `WizardsSection`.
* Email reseting added to Transactional Emails & Newspack Settings > Emails.
* Recaptcha payload changes to accommodate `newspack_recaptcha_credentials ` option.

### How to test the changes in this Pull Request:

1. Checkout and build assets `git checkout feat/epic-ia-trunk-rasacc && npm run build`
2. Test email reseting:
   * Audience >  Setup > Transactional Emails
      * Navigate to _/wp-admin/admin.php?page=newspack-audience#/transactional-emails_
      * Pick an email and take note of the post id associated with it. This can be done by hovering over the "Edit" link.
      * Click "Reset", once request has completed, confirm the email has been reset (I did this by observing that the post id on the "Edit" link had changed).
![Screenshot 2024-12-04 at 14 49 31](https://github.com/user-attachments/assets/223cf422-7cba-436e-906f-e1f0aed744c6)
   * Newspack > Settings > Emails:
      * Navigate to _/wp-admin/admin.php?page=newspack-settings#/emails_
      * Observe "Cancellation" & "Welcome" emails have been added.
      * Follow the above repeatable steps starting at "Pick an email".
3. Navigate to _/wp-admin/admin.php?page=newspack-audience#/payment_
   * Scroll down to "Billing Fields".
       * Confirm newly added "Order notes" checkbox exists.
       * Perform editing & saving of checkbox and confirm changes persist between updates.
   * Scroll down to "Checkout Configuration"
       * Perform editing & saving of toggle and textareas and confirm changes persist between updates.
4. Navigate to _/wp-admin/admin.php?page=newspack-settings&scrollTo=newspack-settings-recaptcha#/_
   * Perform updates to recaptcha settings and confirm data persists between saves.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

adekbadek committed:
Merge pull request #3550 from Automattic/trunk

Alpha release Nov 15

semantic-release-bot committed:
chore(release): 5.8.0-alpha.1 [skip ci]

# [5.8.0-alpha.1](https://github.com/Automattic/newspack-plugin/compare/v5.7.0...v5.8.0-alpha.1) (2024-11-15)

### Bug Fixes

* **reader-registration-block:** fix styles of newsletter lists ([1e1d7a4](https://github.com/Automattic/newspack-plugin/commit/1e1d7a4a0e9945490d83b2f7e29a39dc4b3a09e6))
* **salesforce:** handle woocommerce inactive ([#3521](https://github.com/Automattic/newspack-plugin/issues/3521)) ([bf868a4](https://github.com/Automattic/newspack-plugin/commit/bf868a4b9cc69b7ced5cbe7dd1bb6197f8de6573))
* use user email for esp sync purposes ([#3520](https://github.com/Automattic/newspack-plugin/issues/3520)) ([d6925ff](https://github.com/Automattic/newspack-plugin/commit/d6925ffe8687da0a99ba05ce7ce588e850cd7d43))
* **wc-memberships:** WC for Teams join-team link; team in my-account ([#3540](https://github.com/Automattic/newspack-plugin/issues/3540)) ([489ce0f](https://github.com/Automattic/newspack-plugin/commit/489ce0fb7390f9fabc5956844ac7b437c2c701ac))

### Features

* also trigger subscription change events on switch ([#3514](https://github.com/Automattic/newspack-plugin/issues/3514)) ([5a84949](https://github.com/Automattic/newspack-plugin/commit/5a849497afd67dd73dbf81a8df4d8a68fd8cf53a))
* **pwa:** enable SW file caching ([82af7d4](https://github.com/Automattic/newspack-plugin/commit/82af7d4785e430c6d99b4399ea4e4ed5f1e09a17))
* **starter-content:** e2e improvements; removal CLI command ([35431ac](https://github.com/Automattic/newspack-plugin/commit/35431ac7f1efc4f5048bc511e6b072d0b28e65f0))
* update Stripe Settings dashboard UI ([#3488](https://github.com/Automattic/newspack-plugin/issues/3488)) ([7974fac](https://github.com/Automattic/newspack-plugin/commit/7974faccabff7dc85ce83c9f5d628e0bba20fd9d))
* **wc-memberships:** support team data in import ([63c78b2](https://github.com/Automattic/newspack-plugin/commit/63c78b2a9303b5d0fb342ee27a5560f477cef884))

adekbadek committed:
chore: update FILTER_SANITIZE_STRING to FILTER_SANITIZE_FULL_SPECIAL_CHARS

chickenn00dle committed:
feat(subscriptions): add setting to reattempt payment after final retry (#3560)

This PR adds a new 'On-hold duration' setting for subscriptions behind a feature flag. This setting adds a new retry rule set to the specified number of days in WooCommerce > Settings > Subscriptions > On-hold Duration.

dkoo committed:
Merge pull request #3577 from Automattic/alpha

Release Nov 25

semantic-release-bot committed:
chore(release): 5.8.0 [skip ci]

# [5.8.0](https://github.com/Automattic/newspack-plugin/compare/v5.7.0...v5.8.0) (2024-11-25)

### Bug Fixes

* **reader-registration-block:** fix styles of newsletter lists ([1e1d7a4](https://github.com/Automattic/newspack-plugin/commit/1e1d7a4a0e9945490d83b2f7e29a39dc4b3a09e6))
* **salesforce:** handle woocommerce inactive ([#3521](https://github.com/Automattic/newspack-plugin/issues/3521)) ([bf868a4](https://github.com/Automattic/newspack-plugin/commit/bf868a4b9cc69b7ced5cbe7dd1bb6197f8de6573))
* use user email for esp sync purposes ([#3520](https://github.com/Automattic/newspack-plugin/issues/3520)) ([d6925ff](https://github.com/Automattic/newspack-plugin/commit/d6925ffe8687da0a99ba05ce7ce588e850cd7d43))
* **wc-memberships:** WC for Teams join-team link; team in my-account ([#3540](https://github.com/Automattic/newspack-plugin/issues/3540)) ([489ce0f](https://github.com/Automattic/newspack-plugin/commit/489ce0fb7390f9fabc5956844ac7b437c2c701ac))

### Features

* also trigger subscription change events on switch ([#3514](https://github.com/Automattic/newspack-plugin/issues/3514)) ([5a84949](https://github.com/Automattic/newspack-plugin/commit/5a849497afd67dd73dbf81a8df4d8a68fd8cf53a))
* **pwa:** enable SW file caching ([82af7d4](https://github.com/Automattic/newspack-plugin/commit/82af7d4785e430c6d99b4399ea4e4ed5f1e09a17))
* **starter-content:** e2e improvements; removal CLI command ([35431ac](https://github.com/Automattic/newspack-plugin/commit/35431ac7f1efc4f5048bc511e6b072d0b28e65f0))
* update Stripe Settings dashboard UI ([#3488](https://github.com/Automattic/newspack-plugin/issues/3488)) ([7974fac](https://github.com/Automattic/newspack-plugin/commit/7974faccabff7dc85ce83c9f5d628e0bba20fd9d))
* **wc-memberships:** support team data in import ([63c78b2](https://github.com/Automattic/newspack-plugin/commit/63c78b2a9303b5d0fb342ee27a5560f477cef884))

matticbot committed:
chore(release): merge in release v5.8.0

dkoo committed:
fix: cast values for transaction fee options (#3580)

* fix: cast bool values for transaction fee options

* fix: cast values on front-end, too

semantic-release-bot committed:
chore(release): 5.8.1 [skip ci]

## [5.8.1](https://github.com/Automattic/newspack-plugin/compare/v5.8.0...v5.8.1) (2024-11-26)

### Bug Fixes

* cast values for transaction fee options ([#3580](https://github.com/Automattic/newspack-plugin/issues/3580)) ([82a513f](https://github.com/Automattic/newspack-plugin/commit/82a513fd28d5db9a09de871527bc191923e20957))

matticbot committed:
chore(release): merge in release v5.8.1

matticbot committed:
chore(release): merge in release v5.8.1

leogermani committed:
feat: mark perfmatters as a required plugin (#3578)

adekbadek committed:
feat(woocommerce): remove internal metadata from REST API response

chickenn00dle committed:
feat(ras-acc): add reader account creation and login improvements (#3582)

This PR adds features and styles for the new reader activation and account creation flow.

Co-authored-by: Thomas Guillot <thomasguillot@users.noreply.github.com>
Co-authored-by: Laurel Fulford <laurel.fulford@automattic.com>
Co-authored-by: dkoo <derrick.koo@automattic.com>
Co-authored-by: Adam Cassis <adam.cassis@automattic.com>
Co-authored-by: Derrick Koo <dkoo@users.noreply.github.com>
Co-authored-by: Adam Cassis <adam@adamcassis.com>
Co-authored-by: Miguel Peixe <miguel.peixe@automattic.com>

leogermani committed:
fix: undefined var (#3585)

laurelfulford committed:
fix(ras-acc): make helper text size more specific (#3584)

leogermani committed:
Merge pull request #3586 from Automattic/trunk

Alpha release Nov 29

semantic-release-bot committed:
chore(release): 5.9.0-alpha.1 [skip ci]

# [5.9.0-alpha.1](https://github.com/Automattic/newspack-plugin/compare/v5.8.1...v5.9.0-alpha.1) (2024-11-29)

### Bug Fixes

* **ras-acc:** make helper text size more specific ([#3584](https://github.com/Automattic/newspack-plugin/issues/3584)) ([5bcc688](https://github.com/Automattic/newspack-plugin/commit/5bcc68812937ea979b9d28a08ca95e71fb19a341))
* undefined var ([#3585](https://github.com/Automattic/newspack-plugin/issues/3585)) ([00d8bc7](https://github.com/Automattic/newspack-plugin/commit/00d8bc7f4db78602e948a4ba7dfe5dad2b606672))
* **woocommerce-emails:** use the default email payload if there are no donation products ([#3545](https://github.com/Automattic/newspack-plugin/issues/3545)) ([60c21f3](https://github.com/Automattic/newspack-plugin/commit/60c21f31641a9bbc906d4cc4319fa7d418f8fdf1))

### Features

* mark perfmatters as a required plugin ([#3578](https://github.com/Automattic/newspack-plugin/issues/3578)) ([f20291c](https://github.com/Automattic/newspack-plugin/commit/f20291c7899e99c5d52b8663a10a8e57b7ff1029))
* **ras-acc:** add reader account creation and login improvements ([#3582](https://github.com/Automattic/newspack-plugin/issues/3582)) ([b66de08](https://github.com/Automattic/newspack-plugin/commit/b66de08dcf0a901063577c715257d7ffecd9908d))
* **reader-data:** add a CLI command to align reader membership data ([#3548](https://github.com/Automattic/newspack-plugin/issues/3548)) ([8e49bf0](https://github.com/Automattic/newspack-plugin/commit/8e49bf0a2c9e74958f80de13a52b5fa720e87949))
* **subscriptions:** add setting to reattempt payment after final retry ([#3560](https://github.com/Automattic/newspack-plugin/issues/3560)) ([553c3ac](https://github.com/Automattic/newspack-plugin/commit/553c3ac4327691b3f1826a7de6e3bd5f3d335ccf))
* **woocommerce:** remove internal metadata from REST API response ([6b659a6](https://github.com/Automattic/newspack-plugin/commit/6b659a6afb38d235cf36e116a655352523a09725))

dependabot[bot] committed:
chore(deps-dev): bump @wordpress/browserslist-config

Bumps [@wordpress/browserslist-config](https://github.com/WordPress/gutenberg/tree/HEAD/packages/browserslist-config) from 6.12.0 to 6.13.0.
- [Release notes](https://github.com/WordPress/gutenberg/releases)
- [Changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/browserslist-config/CHANGELOG.md)
- [Commits](https://github.com/WordPress/gutenberg/commits/@wordpress/browserslist-config@6.13.0/packages/browserslist-config)

---
updated-dependencies:
- dependency-name: "@wordpress/browserslist-config"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

laurelfulford committed:
fix(ras-acc): correct My Account custom font sizing clash (#3588)

chickenn00dle committed:
fix(emails): account for false order value (#3590)

This PR fixes an issue where it was possible to call `get_id` on `false` in some edge cases.

laurelfulford committed:
Merge pull request #3591 from Automattic/trunk

Alpha release Dec 02

semantic-release-bot committed:
chore(release): 5.9.0-alpha.2 [skip ci]

# [5.9.0-alpha.2](https://github.com/Automattic/newspack-plugin/compare/v5.9.0-alpha.1...v5.9.0-alpha.2) (2024-12-02)

### Bug Fixes

* **emails:** account for false order value ([#3590](https://github.com/Automattic/newspack-plugin/issues/3590)) ([a2e4042](https://github.com/Automattic/newspack-plugin/commit/a2e404287c7bdeba44689b1a98df9644322d4a71))
* **ras-acc:** correct My Account custom font sizing clash ([#3588](https://github.com/Automattic/newspack-plugin/issues/3588)) ([080f1ce](https://github.com/Automattic/newspack-plugin/commit/080f1ce9e3ac102bdc475b1e5fac4016d48dee2a))

chickenn00dle committed:
feat(woocommerce-subscriptions): add url redirect for wc subscription renewals (#3525)

This PR adds handling for a new `my-account/renew-subscription` endpoint to redirect readers to the first available pending renewal checkout, otherwise the subscriptions my account page.

jaredrethman committed:
Merge remote-tracking branch 'origin/trunk' into feat/epic-ia-trunk-rasacc

jaredrethman committed:
fix: conflicts fixes

jaredrethman committed:
feat: formatted recaptcha

jaredrethman committed:
feat: incorporate new `credentials` property

jaredrethman committed:
refactor: move credentials update to function

jaredrethman committed:
fix: ci / eslint

jaredrethman committed:
Merge remote-tracking branch 'origin/trunk' into feat/epic-ia-trunk-rasacc

adekbadek committed:
feat(wc): duplicate orders admin notice (#3555)

* feat(wc): duplicate orders admin notice

* feat: handle HPOS

* feat: handle order series dismissal

* feat: better UX with a details element

* feat: iterate orders instead of using a direct DB query

* chore: unify wording

* fix: handle the order meta

* feat: CLI tool

* feat: process orders in batches

* feat: tweak

chickenn00dle reviewed (changes requested):
Great job parsing through all the conflicts @jaredrethman!

I tested the RAS-ACC flows (both as admin and as reader) and everything is working for the most part. I just ran into one issue with recaptcha. As admin, changing the recaptcha version causes the following browser error:

```
12:59:32.121 Uncaught TypeError: g is undefined
    pe https://site.test/wp-content/plugins/newspack-plugin/dist/newspack-wizards.js:26
    renderWithHooks https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:15496
    updateFunctionComponent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:19627
    beginWork https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:21650
    beginWork$1 https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:27475
    performUnitOfWork https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26606
    workLoopSync https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26515
    renderRootSync https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26483
    performSyncWorkOnRoot https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26134
    flushSyncCallbacks https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:12052
    flushSyncCallbacksOnlyInLegacyMode https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:12031
    batchedUpdates$1 https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26197
    batchedUpdates https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:3978
    dispatchEventForPluginEventSystem https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:9297
    dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6475
    dispatchEvent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6467
    dispatchDiscreteEvent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6440
newspack-wizards.js:26:5995
```
---

I also ran into a few issues that also exist on the `epic/ia` branch so arent blockers.

When I edit any email template, then go back via the newspack logo, I am directed to `/wp-admin/admin.php?page=newspack` and am shown the following error:

![Screenshot 2024-12-05 at 12 32 55](https://github.com/user-attachments/assets/7e688924-b59d-486f-bc65-30dcf8c81c23)

There is no loading state when saving billing fields:

https://github.com/user-attachments/assets/b1452a0b-74f7-474b-a4e1-c4afa1cc2f27

miguelpeixe commented on src/wizards/audience/views/setup/payment.js:
```diff
@@ -2,28 +2,118 @@
 import BillingFields from '../../components/billing-fields';
 
-export default withWizardScreen( function () {
+export default withWizardScreen( function ( {
+	config,
+	getSharedProps,
+	updateConfig,
+	saveConfig,
+} ) {
```
Suggestion: do you think it'd be too big of a refactor to move this to use the wizard store, as it's being used for 
`newspack-audience/payment`?

It allows for the component to not have to pass down all these props and the "Checkout Configuration" wizard section can become its own standalone component, like `<PaymentGateways />` and the others.

Using the store doesn't seem too different from what it's doing [here](https://github.com/Automattic/newspack-plugin/blob/13d6d55f6a3949ed17118a98369525cfc12ab442/src/wizards/audience/views/setup/index.js#L33-L66):

```js
const data = Wizard.useWizardData( 'newspack-audience/reader-activation' );

updateWizardSettings( {
	slug: 'newspack-audience/reader-activation'
} );

saveWizardSettings( {
	slug: 'newspack-audience/reader-activation'
} );
```

jaredrethman replied:
@miguelpeixe thank you for the review and suggestion. See refactor(s) in 58020abef2b88a5f5430a01173616ef629c3b852, c5f5e6110d7cd68a898e4eb682718ab5577d9897, 3108cde1e004fad558d00cc6216d52ee3382bb2f, 0f6f5cef993f83e1792fde4b84c827a289155a23

miguelpeixe replied:
Thank you for the refactor, @jaredrethman! It works great.

jaredrethman committed:
fix: consistent recaptcha payload

jaredrethman committed:
chore: package-lock

jaredrethman committed:
feat: #pullrequestreview-2482526196

jaredrethman commented:
@chickenn00dle thank you for the review and feedback, some responses below.

> I tested the RAS-ACC flows (both as admin and as reader) and everything is working for the most part. I just ran into one issue with recaptcha. As admin, changing the recaptcha version causes the following browser error:
> ```
> 12:59:32.121 Uncaught TypeError: g is undefined
>     pe https://site.test/wp-content/plugins/newspack-plugin/dist/newspack-wizards.js:26
>     renderWithHooks https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:15496
>     updateFunctionComponent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:19627
>     beginWork https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:21650
>     beginWork$1 https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:27475
>     performUnitOfWork https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26606
>     workLoopSync https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26515
>     renderRootSync https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26483
>     performSyncWorkOnRoot https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26134
>     flushSyncCallbacks https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:12052
>     flushSyncCallbacksOnlyInLegacyMode https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:12031
>     batchedUpdates$1 https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26197
>     batchedUpdates https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:3978
>     dispatchEventForPluginEventSystem https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:9297
>     dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6475
>     dispatchEvent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6467
>     dispatchDiscreteEvent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6440
> newspack-wizards.js:26:5995
> ```

I wasn't able to reproduce this, see below. Maybe `npm install --legacy-peer-deps && npm run build`?

https://github.com/user-attachments/assets/505ee94e-b0c9-4881-b546-0fbe6e8f9c31

> I also ran into a few issues that also exist on the `epic/ia` branch so arent blockers.
> 
> When I edit any email template, then go back via the newspack logo, I am directed to `/wp-admin/admin.php?page=newspack` and am shown the following error:
> 
> ![Screenshot 2024-12-05 at 12 32 55](https://private-user-images.githubusercontent.com/17905991/392929511-7e688924-b59d-486f-bc65-30dcf8c81c23.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MzQ5NjMsIm5iZiI6MTczMzQzNDY2MywicGF0aCI6Ii8xNzkwNTk5MS8zOTI5Mjk1MTEtN2U2ODg5MjQtYjU5ZC00ODZmLWJjNjUtMzBkY2Y4YzgxYzIzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDIxMzc0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExZGMzYjEyNThiMmE3NDAyYzBkNWE5ZTgyYzEyODEwNzgxNDUyMTFjNDVlZjExMmU2NjlkNGQ4MDgyODUyMGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xhAEYzbzZMAHLnVJi-TOmoPUi6e1_-uYKb5jg22nYus)

Can you share some more details here please. I can't seem to find a "newspack logo" when editing an email template

![Screenshot 2024-12-05 at 22 16 14](https://github.com/user-attachments/assets/7e66cc74-93d1-4a39-ac9f-6cfc2d3b318c)

> There is no loading state when saving billing fields:
> 
>  Screen.Recording.2024-12-05.at.12.51.35.mov

This is a trade-off when using `Wizard.useWizardData`, loading states need to be handled on a per/component bases. I've added some visual feedback in ca290e1545d31becb749ec4159bc43008dd27057.

https://github.com/Automattic/newspack-plugin/blob/d2ce4ffc321bb87cd511fe882cadb50605656253/src/wizards/audience/components/billing-fields/index.tsx#L14-L17

https://github.com/user-attachments/assets/cc72cf9b-8ec1-4aa6-9ff8-ce368178c8f5

jaredrethman committed:
fix: ci / eslint

jaredrethman committed:
feat: #discussion_r1872015997 add rest and callbacks

jaredrethman committed:
feat: #discussion_r1872015997 move to component

jaredrethman committed:
feat: #discussion_r1872015997 cleanup payment component

jaredrethman committed:
refactor: is-fetching styles #discussion_r1872015997

jaredrethman committed:
fix: ci / eslint

chickenn00dle commented:
Thanks for the quick response @jaredrethman!

I'll have another look at building a bit later, but in the meantime:

> Can you share some more details here please. I can't seem to find a "newspack logo" when editing an email template

Sorry about that. I meant to say site logo. Its in the top left corner to return to the settings page:
![Screenshot 2024-12-06 at 12 20 59](https://github.com/user-attachments/assets/419896db-0fca-4c20-a4c7-9d5f7810debf)

jaredrethman committed:
feat: replace `newspack` slug with `newspack-dashboard`

jaredrethman commented:
> Thanks for the quick response @jaredrethman!
> 
> I'll have another look at building a bit later, but in the meantime:
> 
> > Can you share some more details here please. I can't seem to find a "newspack logo" when editing an email template
> 
> Sorry about that. I meant to say site logo. Its in the top left corner to return to the settings page: ![Screenshot 2024-12-06 at 12 20 59](https://private-user-images.githubusercontent.com/17905991/393345884-419896db-0fca-4c20-a4c7-9d5f7810debf.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1MDc3NjEsIm5iZiI6MTczMzUwNzQ2MSwicGF0aCI6Ii8xNzkwNTk5MS8zOTMzNDU4ODQtNDE5ODk2ZGItMGZjYS00YzIwLWE0YzctOWQ1Zjc4MTBkZWJmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA2VDE3NTEwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA4MTY2MzY4YmU4NTgwM2E1ZTc5MTI5YTliZTk5NDcxZTlhYjVmY2VjOGFjMDc4YWE3YWEwOGRlZmFjYmMxODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.x9MehJ3TawyOUfPwUxdw3B97_wE0w4d-I3LT-EQhS7o)

Thanks for the screenshot. ff0ffd367da39257da7ec910a55c8a2db6d9f605 fixes the bug!

chickenn00dle committed:
feat(subscriptions): add cancellation reason metadata (#3568)

This PR adds meta data to cancelled woo subscriptions and uses this new meta to sync cancellation reason to ESPs for cancelled subscriptions.

chickenn00dle reviewed (changes requested):
Just tested this again and am still running into issues after reinstall with legacy deps and building. The error seems to be coming from this file: https://github.com/Automattic/newspack-plugin/blob/ff0ffd367da39257da7ec910a55c8a2db6d9f605/src/wizards/newspack/views/settings/connections/recaptcha.tsx#L276

Looks like `settingsToUpdate` initializes with only v3 settings for me (likely because I emptied my v2 keys a while back) and so switching from v3 to v2 causes `versionCredentials` to be undefined. Guarding against this for both `site_key` and `site_secret` fixes the error for me: `versionCredentials?.site_key`/`versionCredentials?.site_secret`.

Although I then see a new issue where the inputs are no longer updating.

jaredrethman committed:
feat: apply default on missing keys

jaredrethman commented:
> Just tested this again and am still running into issues after reinstall with legacy deps and building. The error seems to be coming from this file:
> 
> https://github.com/Automattic/newspack-plugin/blob/ff0ffd367da39257da7ec910a55c8a2db6d9f605/src/wizards/newspack/views/settings/connections/recaptcha.tsx#L276
> 
> Looks like `settingsToUpdate` initializes with only v3 settings for me (likely because I emptied my v2 keys a while back) and so switching from v3 to v2 causes `versionCredentials` to be undefined. Guarding against this for both `site_key` and `site_secret` fixes the error for me: `versionCredentials?.site_key`/`versionCredentials?.site_secret`.
> 
> Although I then see a new issue where the inputs are no longer updating.

Fixed in 61c9e1525a4fcc5c6250fe6e78a3a4e0782e5e6b. Change is mainly to keep the UI code clean, and making the API more consistent supports this better.

miguelpeixe committed:
feat(webhooks): filter request priority (#3587)

laurelfulford committed:
fix(ras-acc): re-add recaptcha to the WooCommerce checkout (#3605)

laurelfulford committed:
fix(ras-acc): re-add recaptcha to the WooCommerce checkout (#3605)

semantic-release-bot committed:
chore(release): 5.9.0-alpha.3 [skip ci]

# [5.9.0-alpha.3](https://github.com/Automattic/newspack-plugin/compare/v5.9.0-alpha.2...v5.9.0-alpha.3) (2024-12-09)

### Bug Fixes

* **ras-acc:** re-add recaptcha to the WooCommerce checkout ([#3605](https://github.com/Automattic/newspack-plugin/issues/3605)) ([07f46b3](https://github.com/Automattic/newspack-plugin/commit/07f46b3bfd7581a6e1e05490e1b2df06677ae59e))

chickenn00dle committed:
feat(wcs): add expired subscription cli tool (#3593)

This PR adds a cli command to migrate the status of all on-hold subscriptions that have failed all retries to expired

dkoo committed:
fix(esp-sync): schedule second sync upon subscription reactivation, just in case (#3603)

* fix(esp-sync): schedule second sync upon subscription reactivation, just in case

* test: add unit test for scheduled sync

* Revert "test: add unit test for scheduled sync"

This reverts commit ee581ed1ee34203e847659600fcc3c45ff1d5cb2.

* feat: if passed data, log that, too

* fix: descended method

* refactor: scheduled sync upon subscription renewal via Data_Events

* refactor: update name of subscription_renewed data event, for clarity

* fix: remove unintentional changes

* docs: inline comment for clarification

Co-authored-by: leogermani <leogermani@automattic.com>

---------

Co-authored-by: leogermani <leogermani@automattic.com>

semantic-release-bot committed:
chore(release): 5.8.2 [skip ci]

## [5.8.2](https://github.com/Automattic/newspack-plugin/compare/v5.8.1...v5.8.2) (2024-12-09)

### Bug Fixes

* **esp-sync:** schedule second sync upon subscription reactivation, just in case ([#3603](https://github.com/Automattic/newspack-plugin/issues/3603)) ([9334295](https://github.com/Automattic/newspack-plugin/commit/93342955db2f70afa9feac6eb8fa349f3703fd72))

matticbot committed:
chore(release): merge in release v5.8.2

dkoo committed:
Merge branch 'alpha' into release

semantic-release-bot committed:
chore(release): 5.9.0 [skip ci]

# [5.9.0](https://github.com/Automattic/newspack-plugin/compare/v5.8.2...v5.9.0) (2024-12-09)

### Bug Fixes

* **emails:** account for false order value ([#3590](https://github.com/Automattic/newspack-plugin/issues/3590)) ([a2e4042](https://github.com/Automattic/newspack-plugin/commit/a2e404287c7bdeba44689b1a98df9644322d4a71))
* **ras-acc:** correct My Account custom font sizing clash ([#3588](https://github.com/Automattic/newspack-plugin/issues/3588)) ([080f1ce](https://github.com/Automattic/newspack-plugin/commit/080f1ce9e3ac102bdc475b1e5fac4016d48dee2a))
* **ras-acc:** make helper text size more specific ([#3584](https://github.com/Automattic/newspack-plugin/issues/3584)) ([5bcc688](https://github.com/Automattic/newspack-plugin/commit/5bcc68812937ea979b9d28a08ca95e71fb19a341))
* **ras-acc:** re-add recaptcha to the WooCommerce checkout ([#3605](https://github.com/Automattic/newspack-plugin/issues/3605)) ([07f46b3](https://github.com/Automattic/newspack-plugin/commit/07f46b3bfd7581a6e1e05490e1b2df06677ae59e))
* undefined var ([#3585](https://github.com/Automattic/newspack-plugin/issues/3585)) ([00d8bc7](https://github.com/Automattic/newspack-plugin/commit/00d8bc7f4db78602e948a4ba7dfe5dad2b606672))
* **woocommerce-emails:** use the default email payload if there are no donation products ([#3545](https://github.com/Automattic/newspack-plugin/issues/3545)) ([60c21f3](https://github.com/Automattic/newspack-plugin/commit/60c21f31641a9bbc906d4cc4319fa7d418f8fdf1))

### Features

* mark perfmatters as a required plugin ([#3578](https://github.com/Automattic/newspack-plugin/issues/3578)) ([f20291c](https://github.com/Automattic/newspack-plugin/commit/f20291c7899e99c5d52b8663a10a8e57b7ff1029))
* **ras-acc:** add reader account creation and login improvements ([#3582](https://github.com/Automattic/newspack-plugin/issues/3582)) ([b66de08](https://github.com/Automattic/newspack-plugin/commit/b66de08dcf0a901063577c715257d7ffecd9908d))
* **reader-data:** add a CLI command to align reader membership data ([#3548](https://github.com/Automattic/newspack-plugin/issues/3548)) ([8e49bf0](https://github.com/Automattic/newspack-plugin/commit/8e49bf0a2c9e74958f80de13a52b5fa720e87949))
* **subscriptions:** add setting to reattempt payment after final retry ([#3560](https://github.com/Automattic/newspack-plugin/issues/3560)) ([553c3ac](https://github.com/Automattic/newspack-plugin/commit/553c3ac4327691b3f1826a7de6e3bd5f3d335ccf))
* **woocommerce:** remove internal metadata from REST API response ([6b659a6](https://github.com/Automattic/newspack-plugin/commit/6b659a6afb38d235cf36e116a655352523a09725))

matticbot committed:
chore(release): merge in release v5.9.0

jaredrethman committed:
Merge remote-tracking branch 'origin/trunk' into feat/epic-ia-trunk-rasacc

chickenn00dle reviewed (approved):
Sorry for the delay on this! All good on my end 👍

miguelpeixe reviewed (approved):
Thank you for the revisions! It's working well and I couldn't find any regressions in other wizards 🎉
